stock_request_template/models/stock_request_template.py

```diff
@@ -78,41 +74,32 @@ class StockRequestTemplateLine(models.Model):
     _description = "stock.request.template.line"
 
     template_id = fields.Many2one("stock.request.template", required=True)
-    product_uom_id = fields.Many2one(
-        "uom.uom",
-        "Product Unit of Measure",
+    product_id = fields.Many2one(
+        "product.product",
+        "Product",
+        domain=[("type", "in", ["product", "consu"])],
+        ondelete="restrict",
         required=True,
-        default=lambda self: self._context.get("product_uom_id", False),
     )
+    product_uom_id = fields.Many2one(
+        "uom.uom", "Product Unit of Measure", required=True,
+    )
+
     product_uom_qty = fields.Float(
         "Quantity",
-        digits=dp.get_precision("Product Unit of Measure"),
+        digits="Product Unit of Measure",
         required=True,
         help="Quantity, specified in the unit of measure indicated in the "
         "request.",
     )
-    product_id = fields.Many2one(
-        "product.product",
-        "Product",
-        domain=[("type", "in", ["product", "consu"])],
-        ondelete="restrict",
-        required=True,
-    )
 
-    @api.constrains("product_id")
-    def _check_product_uom(self):
```

Comment:
why did you remove this?

Reply:
Because I though it was not necessary. The uom field has already a domain that only lets the user choose between the uoms of the product uom category, so the user can not put it wrong. Please correct If I am wrong!